### DISCUSSION TITLE
UiBasedCollector prevents notifications from being sent to Wear OS smart watch

### DISCUSSION THREAD

**Discussion by charludo** [NightscoutFoundation/xDrip#2798] [General]
Hi,

I've encountered the following issue: If my either my phone and watch disconnect, or enough time passes even without any connection loss, notifications stop being sent to the watch (including, but of course not only, BG alarms).

The issue is immediately resolved if I either force close xfrip+, or go into the phone's special access settings and revoke its UiBasedCollector's Notification Access permissions.

I'm just guessing, but I assume xdrip reads the notifications "first", and somhow they then don't end up being read by the Pixel Watch app (that requires the same Access Notifications setting).

More info:
- xdrip+ version: `344deb4-2022.11.16`
- Android version: 10
- phone: Xiaomi Mi Mix 3
- watch: Google Pixel Watch

Unfortunately I have no Android/Java dev experience, or I'd at least try and find a solution myself :(

**Comment by Navid200** [comment]
What is the hardware data source set to?  https://navid200.github.io/xDrip/docs/HardwareDataSource.html

**Reply by charludo** [reply]
The hardware data source is (was) set to "Companion App" (in this case, the patched G7 app). Maybe my original post wasn't quite clear: everything concerning xdrip+ worked flawlessly. But using the UIBasedController/Companion App mode prevents *other* Apps (in this case the Google Pixel app) from reading any notifications (not just xdrip+ notifications - *any* notifications).

I've since switched to a recent nightly build and am using the new (fantastic!) G7 hardware data source. As part of that switch, I have revoked the Notification Access permission from UIBasedController, and the issue has completely disappeared, somewhat confirming the suspicion that it is the root of the issue.
